doc/api/esm.md

```diff
@@ -348,26 +351,24 @@ changes:
 
 * `specifier` {string} The module specifier to resolve relative to the
   current module.
-* Returns: {string} The absolute (`file:`) URL string for the resolved module.
+* Returns: {string} The absolute URL string that the specifier would resolve to.
 
 [`import.meta.resolve`][] is a module-relative resolution function scoped to
 each module, returning the URL string.
 
 ```js
 const dependencyAsset = import.meta.resolve('component-lib/asset.css');
 // file:///app/node_modules/component-lib/asset.css
+import.meta.resolve('./dep');
+// file:///app/dep
 ```
 
 All features of the Node.js module resolution are supported. Dependency
 resolutions are subject to the permitted exports resolutions within the package.
 
-```js
-import.meta.resolve('./dep', import.meta.url);
-// file:///app/dep
-```
-
 > **Caveat** This can result in synchronous file-system operations, which
-> can impact performance similarly to `require.resolve`.
+> can impact performance similarly to `require.resolve`. This feature is not
+> available within custom loaders (it would create a deadlock).
```

Comment:
I think this deserves its own callout and shouldn't be lumped together with an unrelated caveat.